core/src/rpc_subscriptions.rs

```diff
@@ -726,51 +726,75 @@ pub(crate) mod tests {
         } = create_genesis_config(100);
         let bank = Bank::new(&genesis_config);
         let blockhash = bank.last_blockhash();
-        let bank_forks = Arc::new(RwLock::new(BankForks::new(0, bank)));
+        let mut bank_forks = BankForks::new(0, bank);
         let alice = Keypair::new();
-        let tx = system_transaction::transfer(&mint_keypair, &alice.pubkey(), 20, blockhash);
-        let signature = tx.signatures[0];
+
+        let past_bank_tx =
+            system_transaction::transfer(&mint_keypair, &alice.pubkey(), 1, blockhash);
         let unprocessed_tx =
-            system_transaction::transfer(&mint_keypair, &alice.pubkey(), 10, blockhash);
-        let not_ready_signature = unprocessed_tx.signatures[0];
+            system_transaction::transfer(&mint_keypair, &alice.pubkey(), 2, blockhash);
+        let processed_tx =
+            system_transaction::transfer(&mint_keypair, &alice.pubkey(), 3, blockhash);
+
         bank_forks
-            .write()
-            .unwrap()
             .get(0)
             .unwrap()
-            .process_transaction(&tx)
+            .process_transaction(&past_bank_tx)
+            .unwrap();
+
+        let next_bank =
+            Bank::new_from_parent(&bank_forks.banks[&0].clone(), &Pubkey::new_rand(), 1);
+        bank_forks.insert(next_bank);
+
+        bank_forks
+            .get(1)
+            .unwrap()
+            .process_transaction(&processed_tx)
             .unwrap();
 
         let (subscriber, _id_receiver, transport_receiver) =
             Subscriber::new_test("signatureNotification");
-        let sub_id = SubscriptionId::Number(0 as u64);
-        let remaining_sub_id = SubscriptionId::Number(1 as u64);
-        let sink = subscriber.assign_id(sub_id.clone()).unwrap();
+        let id1 = SubscriptionId::Number(1 as u64);
+        let id2 = SubscriptionId::Number(2 as u64);
+        let id3 = SubscriptionId::Number(3 as u64);
+        let sink = subscriber.assign_id(SubscriptionId::Number(0)).unwrap();
         let exit = Arc::new(AtomicBool::new(false));
         let subscriptions = RpcSubscriptions::new(&exit);
-        subscriptions.add_signature_subscription(&signature, None, &sub_id, &sink.clone());
         subscriptions.add_signature_subscription(
-            &not_ready_signature,
-            None,
-            &remaining_sub_id,
+            &past_bank_tx.signatures[0],
```

Comment:
What about adding an additional subscription on `past_bank_tx.signatures[0]` with confirmations `Some(1)` to test a non-zero case?

Reply:
Great call, added